server/src/main/java/org/apache/druid/segment/realtime/firehose/HttpFirehoseFactory.java

```diff
@@ -46,6 +47,8 @@
   private static final Logger log = new Logger(HttpFirehoseFactory.class);
   private final List<URI> uris;
   private final boolean supportContentRange;
+  private final String httpAuthenticationUsername;
+  private final String httpAuthenticationPassword;
```

Comment:
This will be printed in the task log if it's string. Please use `PasswordProvider` to make it configurable.

Reply:
Thank you for the comment. Could you elaborate a bit on what you mean by configurable? Once wrapped in a PasswordProvider, won't the content still need to be serialized in plaintext for it to be passed between services in the cluster (overlord --> worker)? It is not clear to me how we could selectively mask the password just by encapsulating it within a password provider. I've been trying to run a search on the codebase to find a similar situation but have come up empty so far.

Any tips are appreciated!